### ISSUE TITLE
Disconnected Foreach - severity changes are not applied

### ISSUE THREAD

**Issue by funivan** [kalessil/phpinspectionsea#268]
How to reproduce. Get latest syfmony version and write this code: 
```php
$bar = new \Symfony\Component\Console\Helper\ProgressBar($output);
$items = [1, 2, 3];
foreach($items as $i) {
    $bar->advance(); // <- This statement seems to be disconnected ...
    echo $i;
}
```

`$bar->advance()` -- should not be disconnected

Also I can`t change severity of this inspection.

**Comment by kalessil** [comment]
Severity: will check why.
False-positive: nope - reported as needed. Implementation needs enhancement, e.g. you can use setProgress/setProgress instead to make inner operation really dependent on the loop variables.

**Comment by funivan** [comment]
@kalessil in some cases I agree with you. But http://symfony.com/doc/current/components/console/helpers/progressbar.html we have ` $progress->advance();`  so a lot of developers using symfony and this plugin get errors.
P.S. Write `$bar->setProgress($bar->getProgress()+1);` is less c lean ;)

**Comment by kalessil** [comment]
I meant something like this:
```php
$bar = new \Symfony\Component\Console\Helper\ProgressBar();
$initialProgress = $bar->getProgress();
foreach([1, 2, 3] as $i) { // or perhaps for loop in this case
    $bar->setProgress($i + $initialProgress);
}
```

**Comment by funivan** [comment]
This was the sample cases. Here is real one:
```php
  $bar = new ProgressBar($output);
  $dealers = ...; // array of objects
  foreach ($dealers as $dealer) {
    $bar->advance();
    mail($dealer->email, 'subject', 'body');
 }
```

**Comment by kalessil** [comment]
This already something to work with:

```php
$bar = new ProgressBar($output);
  $dealers = ...; // array of objects
  foreach ($dealers as $dealer) {
    if (mail($dealer->email, 'subject', 'body')) {
        $bar->advance();
    } 
    // and here some logging is asking to be added, but I'm not aware of wider context
 }
```

This way you progression is really based on the business logic: sending e-mail, was it really send.

**Comment by kalessil** [comment]
@Garethp : you can pick up this issue

The fix is to replace ProblemHighlightType.WEAK_WARNING with ProblemHighlightType.GENERIC_ERROR_OR_WARNING and ensure that in plugin.xml the inspection error level is the weak warning.

**Comment by kalessil** [comment]
@funivan : what do you think about the last code fragment from me?

**Comment by funivan** [comment]
Ok. I get your point) But:
1. Sometimes we have function that return nothing. Instead they fire exception. For example:
```
$dealer->setBalance(10);
$dealer->save(); // save does not return any value
```
I do not know how to deal in this situation with `advanced()`. (Yep i can use `setProgress`)

2. What to do if i want to iterate over all items, display progress of iteration, and send email to only 10-100 dealers. I have 100000 dealers for example.

**Comment by kalessil** [comment]
1) Wrap `save` with try-catch, handle exception - the send out should not stop on the single failure.
2) Depends on the progress bar intention: is it for human?

**Comment by funivan** [comment]
1. ok
2. yes for human. Bar has `message` and we can put number of processed items: `$bar->setMessage($itemsNum)` Sow it is readable and usable for developers)

**Comment by Garethp** [comment]
@kalessil Can you assign this to me? Or add me to the repository as someone who can do that?

**Comment by kalessil** [comment]
@Garethp : you should receive an invitation to join collaborators group. Feel free to re-assign the ticket.

**Comment by kalessil** [comment]
@funivan : 2 looks damn great, you can actually try to generate a message with both success and failure numbers.

And if we step back - that spin off (really great IMO) of the inspection complains.
